Tests for rewrite hellper

TorkelE opened the pull request.

TorkelE commented:
With some exceptions ("LuxCore extensions Test" and "Groebner extension Test", which should be unrelated to this), tests passes locally. I think CI can be activated for the PR now (@shashi).

ChrisRackauckas commented on src/rewrite-helpers.jl:
```diff
@@ -0,0 +1,134 @@
+"""
```
We should probably move `ModelingToolkit.fixedpoint_sub` here as well?

ChrisRackauckas commented:
Missing a docs page.

@YingboMa should review the implementation.

TorkelE commented:
I have added entries in the docs now. `Symbolics.occursin` and `Symbolics.filterchildren` did not have very natural places, so I added these with `Symbolics.replace`.

YingboMa reviewed (approved):
LGTM
